node/test/toolrunnertests.ts

```diff
@@ -395,8 +395,68 @@ describe('Toolrunner Tests', function () {
                 done(err);
             });
     })
+    // function to compile a .NET program that prints lines.
+    // the helper program is used on Windows to validate piping output between tools.
+    let compileOutputExe = (): string => {
+        let directory = path.join(testutil.getTestTemp(), 'print-output-exe');
+        tl.mkdirP(directory);
+        let exePath = path.join(directory, 'print-output.exe');
+
+        // short-circuit if already compiled
+        try {
+            fs.statSync(exePath);
+            return exePath;
+        }
+        catch (err) {
+            if (err.code != 'ENOENT') {
+                throw err;
+            }
+        }
+
+        let sourceFile = path.join(__dirname, 'scripts', 'print-output-exe.cs');
+        let cscPath = 'C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319\\csc.exe';
+        fs.statSync(cscPath);
+        child_process.execFileSync(
+            cscPath,
+            [
+                '/target:exe',
+                `/out:${exePath}`,
+                sourceFile
+            ]);
+        return exePath;
+    }
+    // function to compile a .NET program that matches input lines.
+    // the helper program is used on Windows to validate piping output between tools.
+    let compileMatchExe = (): string => {
+        let directory = path.join(testutil.getTestTemp(), 'match-input-exe');
+        tl.mkdirP(directory);
+        let exePath = path.join(directory, 'match-input.exe');
+
+        // short-circuit if already compiled
+        try {
+            fs.statSync(exePath);
+            return exePath;
+        }
+        catch (err) {
+            if (err.code != 'ENOENT') {
+                throw err;
+            }
+        }
+
+        let sourceFile = path.join(__dirname, 'scripts', 'match-input-exe.cs');
+        let cscPath = 'C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319\\csc.exe';
+        fs.statSync(cscPath);
+        child_process.execFileSync(
+            cscPath,
+            [
+                '/target:exe',
+                `/out:${exePath}`,
+                sourceFile
+            ]);
+        return exePath;
+    }
     it('Exec pipe output to another tool, succeeds if both tools succeed', function(done) {
-        this.timeout(1000);
+        this.timeout(20000);
```

Comment:
unfortunately calling csc twice and then invoking two processes can take a while sometimes depending upon the machine